vortex-array/src/compute/take.rs

```diff
@@ -10,6 +11,25 @@ pub trait TakeFn {
 pub fn take(array: impl AsRef<Array>, indices: impl AsRef<Array>) -> VortexResult<Array> {
     let array = array.as_ref();
     let indices = indices.as_ref();
+
+    if !indices.dtype().is_int() || indices.dtype().is_nullable() {
+        vortex_bail!(
+            "Take indices must be a non-nullable integer type, got {}",
+            indices.dtype()
+        );
+    }
+
+    // If the indices are large enough, it's faster to canonicalize the array and then take
+    // except for sparse arrays, where patching is faster in that form.
+    if indices.len() < array.len() || SparseArray::try_from(array).is_ok() {
+        do_take(array, indices)
```

Comment:
if we keep this PR, perhaps this should be a `canonicalize_threshold` method on TakeFn